lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -2721,15 +2721,32 @@ _SPIRV_OP(PowN, true, 10)
 
 class SPIRVAtomicInstBase : public SPIRVInstTemplateBase {
 public:
+  llvm::Optional<ExtensionID> getRequiredExtension() const override {
+    if (getOpCode() == OpAtomicFAddEXT)
+      return ExtensionID::SPV_EXT_shader_atomic_float_add;
+    return {};
+  }
+
   SPIRVCapVec getRequiredCapability() const override {
     SPIRVCapVec CapVec;
+    if (!hasType())
+      return CapVec;
+
     // Most of atomic instructions do not require any capabilities
     // ... unless they operate on 64-bit integers.
-    if (hasType() && getType()->isTypeInt(64)) {
+    if (getType()->isTypeInt(64)) {
       // In SPIRV 1.2 spec only 2 atomic instructions have no result type:
       // 1. OpAtomicStore - need to check type of the Value operand
       // 2. OpAtomicFlagClear - doesn't require Int64Atomics capability.
       CapVec.push_back(CapabilityInt64Atomics);
+    } else if (getOpCode() == OpAtomicFAddEXT) {
```

Comment:
```suggestion
    } else if (getOpCode() == OpAtomicFAddEXT) {
      // ... or unless they operate on floating-point types
```

Actually, I'm not entirely sure about this comment, because the `if` might be expanded later to accommodate other atomic instructions from various extensions, so probably we should leave just one, but generic comment, which explains possible reasons why some capabilities might be required?

Reply:
Thanks for bringing this one up! Replaced the comment with a more general one.